Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -489,6 +489,109 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
             return start->to_string(mode);
         return MUST(String::formatted("{} / {}", start->to_string(mode), end->to_string(mode)));
     }
+    case PropertyID::Mask: {
+        StringBuilder builder;
+
+        auto serialize_layer = [mode, &builder](String image_value_string, String position_value_string, String size_value_string, String repeat_value_string, String origin_value_string, String clip_value_string, String composite_value_string, String mode_value_string) {
+            PropertyID canonical_property_order[] = {
+                PropertyID::MaskImage,
+                PropertyID::MaskPosition,
+                // Intentioanlly skipping MaskSize here, it is handled together with MaskPosition.
+                PropertyID::MaskRepeat,
+                PropertyID::MaskOrigin,
+                PropertyID::MaskClip,
+                PropertyID::MaskComposite,
+                PropertyID::MaskMode,
+            };
+
+            auto property_value = [&](PropertyID property) -> String const& {
+                switch (property) {
+                case PropertyID::MaskImage:
+                    return image_value_string;
+                case PropertyID::MaskPosition:
+                    return position_value_string;
+                case PropertyID::MaskSize:
+                    return size_value_string;
+                case PropertyID::MaskRepeat:
+                    return repeat_value_string;
+                case PropertyID::MaskOrigin:
+                    return origin_value_string;
+                case PropertyID::MaskClip:
+                    return clip_value_string;
+                case PropertyID::MaskComposite:
+                    return composite_value_string;
+                case PropertyID::MaskMode:
+                    return mode_value_string;
+                default:
+                    VERIFY_NOT_REACHED();
+                }
+            };
+
+            auto is_initial_value = [mode, property_value](PropertyID property) -> bool {
+                return property_value(property) == property_initial_value(property)->to_string(mode);
+            };
+
+            bool layer_is_empty = true;
+            for (size_t i = 0; i < array_size(canonical_property_order); i++) {
+                auto property = canonical_property_order[i];
+                auto const& value = property_value(property);
+
+                if (is_initial_value(property)) {
+                    if (property != PropertyID::MaskOrigin)
+                        continue;
```

Comment:
I think this will skip serializing `mask-position` if it's the initial value - but that's only safe to do if `mask-size` is also the initial value.